pwndbg/heap/ptmalloc.py

```diff
@@ -464,9 +468,20 @@ def bin_at(self, index, arena_addr=None):
 
         front, back = normal_bins[index * 2], normal_bins[index * 2 + 1]
         fd_offset   = self.chunk_key_offset('fd')
+        bk_offset   = self.chunk_key_offset('bk')
+
+        is_chain_corrupted = False
+        chain_fd = pwndbg.chain.get(int(front), offset=fd_offset, hard_stop=current_base, limit=heap_chain_limit, include_start=True)
+        chain_bk = pwndbg.chain.get(int(back), offset=bk_offset, hard_stop=current_base, limit=heap_chain_limit, include_start=True)
+
+        # check if bin[index] points to itself (is empty)
+        if len(chain_fd) == 2 and len(chain_bk) == 2 and chain_fd[0] == chain_bk[0]:
```

Comment:
as its Python it can be written as `if len(chain_fd) == len(chain_bk) == 2 and ...`; I don't mind leaving it as it is; it's up to you